Mathlib/Analysis/CauchyBound.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Algebra.GeomSum
+import Mathlib.Algebra.Polynomial.Eval
+import Mathlib.Analysis.Normed.Field.Basic
+
+/-!
+# Cauchy's bound on polynomial roots.
+-/
+
+variable {K : Type*} [NormedDivisionRing K]
+
+open Polynomial Finset NNReal
+
+/--
+Cauchy's bound on the roots of a given polynomial.
+-/
+noncomputable def cauchyBound (p : K[X]) : ℝ≥0 :=
+  sup (range p.natDegree) (‖p.coeff ·‖₊) / ‖p.leadingCoeff‖₊
+
+lemma NNReal.geom_sum {x : ℝ≥0} (h : 1 < x) (n : ℕ) :
+    ∑ i ∈ Finset.range n, x ^ i = (x ^ n - 1) / (x - 1) := by
```

Comment:
This should go to `Data/NNReal`

Reply:
It would be bad to import `Algebra.GeomSum` in `Data.NNReal.Basic`, right? Should I make a new file there for this?